src/Extension/FluentExtension.php

```diff
@@ -1176,4 +1197,110 @@ public function updateLocalisationTabConfig(GridFieldConfig $config)
             }
         }
     }
+
+    public function getLocalisedCopyActive(): bool
+    {
+        return $this->localisedCopyActive;
+    }
+
+    public function setLocalisedCopyActive(bool $active): DataObject
+    {
+        $this->localisedCopyActive = $active;
+
+        return $this->owner;
+    }
+
+    /**
+     * Localised copy global state manipulation
+     * useful for disabling localised copy feature in parts of the code
+     *
+     * @param callable $callback
+     * @return mixed
+     */
+    public function withLocalisedCopyState(callable $callback)
+    {
+        $active = $this->localisedCopyActive;
+
+        try {
+            return $callback();
+        } finally {
+            $this->localisedCopyActive = $active;
+        }
+    }
+
+    /**
+     * Duplicate related objects based on configuration
+     * Provides an extension hook for custom duplication
+     */
+    protected function makeLocalisedCopy(): void
+    {
+        if (!$this->localisedCopyNeeded()) {
+            return;
+        }
+
+        $owner = $this->owner;
+        $relations = (array) $owner->config()->get('localised_copy');
+
+        $owner->invokeWithExtensions('onBeforeLocalisedCopy');
+
+        foreach ($relations as $relation => $field) {
+            $original = $owner->{$relation}();
+
+            if (!$original instanceof DataObject) {
+                continue;
+            }
+
+            if (!$original->exists()) {
+                continue;
+            }
+
+            $duplicate = $original->duplicate();
+            $owner->{$field} = $duplicate->ID;
+        }
+
+        $owner->invokeWithExtensions('onAfterLocalisedCopy');
```

Comment:
In case anyone needs it, it wouldn't hurt to add both $original and $duplicate as arguments (in case code needs to adjust / copy fields).

Reply:
Oh right, I forgot it's in a loop. You sorted out a solution anyway nice :)